confmap/converter/expandconverter/expand.go

```diff
@@ -80,21 +80,27 @@ func (c converter) expandStringValues(value any) (any, error) {
 func (c converter) expandEnv(s string) (string, error) {
 	var err error
 	res := os.Expand(s, func(str string) string {
+		// This allows escaping environment variable substitution via $$, e.g.
+		// - $FOO will be substituted with env var FOO
+		// - $$FOO will be replaced with $FOO
+		// - $$$FOO will be replaced with $ + substituted env var FOO
+		if str == "$" {
+			return "$"
+		}
```

Comment:
Can handle this via https://github.com/open-telemetry/opentelemetry-collector/pull/10392 if you want

Reply:
Merged!